imap/sync_log.h

```diff
@@ -73,6 +73,9 @@ void sync_log_channel(const char *channel, const char *fmt, ...);
 #define sync_log_unmailbox(name) \
     sync_log("UNMAILBOX %s\n", name)
 
+#define sync_log_mailbox_double(name1, name2) \
+    sync_log("MAILBOX %s %s\n", name1, name2)
```

Comment:
This is supposed to be "MAILBOX %s\nMAILBOX %s\n" - but since it's a single write under lock, they're guaranteed to be picked up by the same run.

I don't THINK the reader handling multiple mailboxes is safe like this because it will parse as a single name in the reader - whitespace isn't special!

Reply:
This is handled by the change in sync_client.c, also in this PR -- it explicitly looks for a second MAILBOX argument now.

It needs to be like this now, and not two distinct MAILBOX lines like it once was, since we started batching -- we need to know that in this instance, these two mailboxes are related, even if they don't belong to the same user, so that they can be batched and ordered correctly.

This originally happened at 1fa18fb128, but then when I later added the distinct RENAME command I mistakenly removed it (breaking copy/move).  The fix here is bringing back the updated mailbox double for copy/move, while keeping RENAME for renames.

The parser in `sync_log_reader_getitem()` already handles two arguments properly (and looks like it has done so forever).  It looks for a word, an astring, and then an optional second astring...  though this now has me wondering whether `sync_log()` itself _writes_ the correct thing?  I'm skimming quickly, but I don't see it doing any quoting -- looks like it just uses printf formatting?  So I wonder what happens if a mailbox name contains a space??

Reply:
Okay, I've checked out `sync_log()` and it's quoting properly after all, I just missed the salient detail while skimming.

So I think this is all fine?